Refactor target ruby version

pocke opened the pull request:
See. https://github.com/bbatsov/rubocop/pull/3824#discussion_r93827394


-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [ ] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

bbatsov commented on lib/rubocop/cop/mixin/target_ruby_version.rb:
```diff
@@ -0,0 +1,17 @@
+      end
+
+      def support_target_ruby_version?(version)
+        return true unless version
```
I guess you meant to write `unless @minimum_targer_ruby_version`

bbatsov commented on lib/rubocop/cop/mixin/target_ruby_version.rb:
```diff
@@ -0,0 +1,17 @@
+
+module RuboCop
+  module Cop
+    #
```
Seems you forgot to add the documentation here.

pocke commented:
> I guess you meant to write unless @minimum_targer_ruby_version


Hmm, I was confusing.
Both `unless version` and `unless @minimum_target_ruby_version` are not needed.
I removed it.

> Seems you forgot to add the documentation here.

I forgot it. Thanks!




I fixed.

backus commented:
@pocke would it be easier to just have this code in the parent cop class, have the default minimum ruby version be 2.0 for all cops, and then the cops that only apply to more recent ruby versions would do `minimum_ruby_version 2.2`? This way you wouldn't need to mix in a module and do `cop.class.respond_to?(:support_target_ruby_version?)`

bbatsov commented:
This was actually my initial idea, but I'm fine with either approach.

pocke commented:
I'm fine with either approach, too. Is current implementation okay?

bbatsov commented:
Yeah, it's ok. Thanks! 👍
